### PR TITLE
[PodTarget] Ensure the deployment target is high enough for frameworks

### LINKED ISSUE #4552: Deployment target is set incorrectly
Testing on cocoapods master, commit 0f3eb52b3ec9a27635f872b53c5b5f99d5148447

Testing a previous issue, I've noticed that the deployment targets are not being set correctly.
screen shot 2015-11-17 at 12 30 40

The issue is present here in this sample project;
https://github.com/mlloyd/PodExample_tvos_ios_bug

### PR DESCRIPTION
**Pull request by segiddins**
closes #4552.

- [x] Documentation
- [x] ~~CHANGELOG~~ (fixing a bug introduced since 0.39.0)
- [x] Specs
- [x] Rebuild integration specs

This is really really gross, but more correct. Would appreciate suggestions on how to refactor it.

### CODE REVIEW

**Comment by segiddins** [comment]
@CocoaPods/core thoughts on this one, please?

**Comment by neonichu** [comment]
@segiddins I'm not quite sure what this even does :) Could you outline the problem a bit?

**Comment by segiddins** [comment]
Makes sure the iOS deployment target is not less than 8.0 when using frameworks.

**Comment by neonichu** [comment]
I thought that we would already do that, that's what confused me, but turns out it is only something we check when validating the deployment target of a Pod.

File: lib/cocoapods/target/pod_target.rb [hunk 1]
```diff
@@ -88,12 +88,30 @@ def label
       end
     end
 
-    # @return [Platform] the platform for this target.
+    # The deployment target for the pod target, which is the maximum of all
```
Comment (manuyavuz) [inline lib/cocoapods/target/pod_target.rb]: This documentation probably wrongly copied.

File: lib/cocoapods/target/pod_target.rb [hunk 2]
```diff
@@ -88,12 +88,30 @@ def label
...
+    # @return [String] The deployment target.
     #
     def platform
-      @platform ||= target_definitions.first.platform
+      @platform ||= begin
+        platform_name = target_definitions.first.platform.name
+        default = Podfile::TargetDefinition::PLATFORM_DEFAULTS[platform_name]
+        deployment_target = specs.map do |spec|
+          version = Pod::Version.new(spec.deployment_target(platform_name) || default)
+          if platform_name == :ios && requires_frameworks?
```
Comment (manuyavuz) [inline lib/cocoapods/target/pod_target.rb]: You could do this update outside specs.map once after getting max deployment target for given specs.

**Comment by segiddins** [comment]
I don't understand why the integration specs are failing in the way they are?

**Comment by segiddins** [comment]
@CocoaPods/core please review. I will update the integration specs and merge once i get a thumbs-up

**Comment by manuyavuz** [comment]
@segiddins Is the change in integration specs for `install_custom_module_name` normal? User project has deployment target of `8.2`, previous execution output was including `- Installing target 'SocketRocket' iOS 8.2` as expected, but with your changes, it has returned to `- Installing target 'SocketRocket' iOS 8.0`

I'm not sure this is an expected change.

**Comment by segiddins** [comment]
Yes, this is exactly the expected change, since the pod's deployment target (as set in the podspec) is 8.0

**Comment by manuyavuz** [comment]
Could you direct me to the corresponding podspec? I see `ios 6.0` here: https://github.com/CocoaPods/Specs/blob/master/Specs/libPusher/1.6.1/libPusher.podspec.json Does it use a different podspec?

**Comment by segiddins** [comment]
Ah sorry, was thinking about the other spec. No, it uses that podspec, but its integrating as a framework, so the minimum deployment target shouldn't be lower than 8.0

**Comment by manuyavuz** [comment]
But the thing is it was also having `iOS 8.2` before, and downgraded to `8.0`. I think I misunderstand the proposed fix here. Previous to your changes, Pods project was getting deployment target from user project, which was causing wrong deployment target setups, and with your change, deployment targets for Pods project and pod targets are solely determined via constraints in podfile or podspecs Therefore, it correctly downgraded to `8.0` because only requirement for the pod was `use_frameworks!` The `8.2` requirement set by user project should not effect the requirement of Pods project. Am I right?

**Comment by segiddins** [comment]
it being 8.2 earlier was probably a bug. but yes, you're correct that user project deployment targets should no longer effect the deployment targets for podtargets

**Comment by manuyavuz** [comment]
So we could directly observed that your change really ensures the deployment target requirement for pod targets :)

Thumbs up from me!

**Comment by segiddins** [comment]
:shipit:

**Pull request merged**